src/Features/Core/Portable/SolutionCrawler/IDocumentTrackingServiceExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class IDocumentTrackingServiceExtensions
+    {
+        /// <summary>
+        /// Gets the active <see cref="Document"/> the user is currently working in. May be null if
+        /// there is no active document or the active document is not in this <paramref name="solution"/>.
+        /// </summary>
+        public static Document GetActiveDocument(this IDocumentTrackingService service, Solution solution)
+        {
+            // Note: GetDocument checks that the DocId is contained in the solution, and returns null if not.
+            return solution.GetDocument(service.GetActiveDocument());
+        }
+
+        /// <summary>
+        /// Get a read only collection of all the unique visible documents in the workspace that are
+        /// contained within <paramref name="solution"/>.
+        /// </summary>
+        public static ImmutableArray<Document> GetVisibleDocuments(this IDocumentTrackingService service, Solution solution)
+            => service.GetVisibleDocuments()
+                      .Select(d => solution.GetDocument(d))
+                      .WhereNotNull()
+                      .Distinct()
```

Comment:
Since we need to de-dupe anyways, would it make sense to use a set for the result? To get the priority documents, it would just be adding one more document to that set. 
Also, would it make sense to use document IDs rather than documents, since that's what seems to be used during search anyways?

Reply:
So, in teh case where there's an 'active' document **and ** 'visible' documents from the same project, we want the 'active' document to take precedence.  That's why i use a list and not a set.

Reply:
> Also, would it make sense to use document IDs rather than documents, since that's what seems to be used during search anyways?

I've flip/flopped about that.  I think i likke it better talking about the real docs, and not IDs...